electronics_lib/TableDeratingCapacitor.py

```diff
@@ -0,0 +1,190 @@
+from electronics_abstract_parts.Categories import DummyDevice
+from electronics_abstract_parts import *
+from .PartsTable import *
+import math
+from .JlcFootprint import JlcFootprint
+
+class CapacitorTable(LazyTable):
+  CAPACITANCE: PartsTableColumn[Range]
+  NOMINAL_CAPACITANCE: PartsTableColumn[float]
+  VOLTAGE_RATING: PartsTableColumn[Range]
+  FOOTPRINT: PartsTableColumn[str]
+
+  MANUFACTURER: str
+  PART_NUMBER: str
+  DATASHEETS: str
+  DESCRIPTION: str
+  COST: PartsTableColumn[float]
+
+
+class TableDeratingCapacitor(Capacitor, FootprintBlock, GeneratorBlock):
+  _TABLE: Type[CapacitorTable]
+
+  DERATE_VOLTCO = {  # in terms of %capacitance / V over 3.6
+    #  'Capacitor_SMD:C_0603_1608Metric'  # not supported, should not generate below 1uF
+    'Capacitor_SMD:C_0805_2012Metric': 0.08,
+    'Capacitor_SMD:C_1206_3216Metric': 0.04,
+  }
+  DERATE_MIN_VOLTAGE = 3.6  # voltage at which derating is zero
+  DERATE_MIN_CAPACITANCE = 1.0e-6
+  DERATED_CAPACITANCE = PartsTableColumn(Range)
+
+  PARALLEL_COUNT = PartsTableColumn(int)
+  PARALLEL_CAPACITANCE = PartsTableColumn(Range)
+  PARALLEL_DERATED_CAPACITANCE = PartsTableColumn(Range)
+  PARALLEL_COST = PartsTableColumn(float)
+
+  @init_in_parent
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+
+    self.part_spec = self.Parameter(StringExpr(""))
+    self.footprint_spec = self.Parameter(StringExpr(""))
+
+    # Default that can be overridden
+    self.single_nominal_capacitance = self.Parameter(RangeExpr((0, 22)*uFarad))  # maximum capacitance in a single part
+
+    self.generator(self.select_capacitor, self.capacitance, self.voltage, self.single_nominal_capacitance,
+                   self.part_spec, self.footprint_spec)
+
+    # Output values
+    self.selected_capacitance = self.Parameter(RangeExpr())
+    self.selected_derated_capacitance = self.Parameter(RangeExpr())
+    self.selected_voltage_rating = self.Parameter(RangeExpr())
+
+  def select_capacitor(self, capacitance: Range, voltage: Range,
+                       single_nominal_capacitance: Range,
+                       part_spec: str, footprint_spec: str) -> None:
+    # Pre-filter out by the static parameters
+    # Note that we can't filter out capacitance before derating
+    prefiltered_parts = self.filter_capacitor(voltage, single_nominal_capacitance, part_spec, footprint_spec)
+
+    derated_parts = self.add_derated_capacitance(voltage, prefiltered_parts)
+    # If the min required capacitance is above the highest post-derating minimum capacitance, use the parts table.
+    # An empty parts table handles the case where it's below the minimum or does not match within a series.
+
+    derated_max_min_capacitance = max(derated_parts.map(lambda row: row[self.DERATED_CAPACITANCE].lower))
```

Comment:
```suggestion

    # If the min required capacitance is above the highest post-derating minimum capacitance, use the parts table.
    # An empty parts table handles the case where it's below the minimum or does not match within a series.
    derated_max_min_capacitance = max(derated_parts.map(lambda row: row[self.DERATED_CAPACITANCE].lower))
```

Comment should be closer to the relevant code